core/indexer/common.go

```diff
@@ -448,15 +412,62 @@ func prepareTxUpdate(tx *Transaction) ([]byte, []byte) {
 
 	if tx.GasUsed == tx.GasLimit {
 		// do not update gasUsed because it is the same with gasUsed when transaction was saved first time in database
-		serializedData = []byte(fmt.Sprintf(`{ "doc" : { "log" : %s, "scResults" : %s, "status": "%s", "timestamp": %s } }`,
-			string(marshalizedLog), string(scResults), tx.Status, string(marshalizedTimestamp)))
+		serializedData =
+			[]byte(fmt.Sprintf(`{"script":{"source":"`+
+				`ctx._source.status = params.status;`+
+				`ctx._source.log = params.log;`+
+				`ctx._source.scResults = params.scResults;`+
+				`ctx._source.timestamp = params.timestamp;`+
+				`","lang": "painless","params":`+
+				`{"status": "%s", "log": %s, "scResults": %s, "timestamp": %s}},"upsert":%s}`,
+				tx.Status, string(marshalizedLog), string(scResults), string(marshalizedTimestamp), string(marshalledTx)))
 	} else {
 		// update gasUsed because was changed (is a smart contract operation)
-		serializedData = []byte(fmt.Sprintf(`{ "doc" : { "log" : %s, "scResults" : %s, "status": "%s", "timestamp": %s, "gasUsed" : %s } }`,
-			string(marshalizedLog), string(scResults), tx.Status, string(marshalizedTimestamp), fmt.Sprintf("%d", tx.GasUsed)))
+		serializedData =
+			[]byte(fmt.Sprintf(`{"script":{"source":"`+
+				`ctx._source.status = params.status;`+
+				`ctx._source.log = params.log;`+
+				`ctx._source.scResults = params.scResults;`+
+				`ctx._source.timestamp = params.timestamp;`+
+				`ctx._source.gasUsed = params.gasUsed;`+
+				`","lang": "painless","params":`+
+				`{"status": "%s", "log": %s, "scResults": %s, "timestamp": %s, "gasUsed": %d}},"upsert":%s}`,
+				tx.Status, string(marshalizedLog), string(scResults), string(marshalizedTimestamp), tx.GasUsed, string(marshalledTx)))
+	}
+
+	log.Warn("log meta - update", "data", string(meta))
```

Comment:
why do we need 2 log.Warn calls here? Also, srld hits again and you have commented out code.

Reply:
removed